### PR TITLE
Update NVIDIA libs used for OSS builds

### PR DESCRIPTION
**Pull request by nluehr**
Updating to
- CUDA 11.7
- CUDNN 8.4.1
- TensorRT 8.4.1

### CODE REVIEW

File: tf_sig_build_dockerfiles/devel.usertools/gpu.bazelrc [hunk 1]
```diff
@@ -41,12 +41,9 @@ build --@local_config_cuda//:enable_cuda
...
 build --action_env=TF_CUDNN_VERSION="8"
-build --action_env=CUDA_TOOLKIT_PATH="/usr/local/cuda-11.2"
+build --action_env=CUDA_TOOLKIT_PATH="/usr/local/cuda"
 build --action_env=GCC_HOST_COMPILER_PATH="/dt9/usr/bin/gcc"
-# TensorRT 7 for CUDA 11.1 is compatible with CUDA 11.2, but requires
-# libnvrtc.so.11.1. See https://github.com/NVIDIA/TensorRT/issues/1064.
-# TODO(b/187962120): Remove CUDA-11.1 when upgrading to TensorRT 8.
-build --action_env=LD_LIBRARY_PATH="/usr/local/cuda:/usr/local/cuda/lib64:/usr/local/cuda/extras/CUPTI/lib64:/usr/local/tensorrt/lib:/usr/local/cuda-11.1/lib64""
+build --action_env=LD_LIBRARY_PATH="/usr/local/cuda/lib64:/usr/local/cuda/extras/CUPTI/lib64"
 build --crosstool_top=@ubuntu20.04-gcc9_manylinux2014-cuda11.2-cudnn8.1-tensorrt7.2_config_cuda//crosstool:toolchain
```
Comment (nluehr) [inline tf_sig_build_dockerfiles/devel.usertools/gpu.bazelrc:47]: It seems that this line should also be updated, but a CUDA 11.7, cudnn 8.4, TRT 8.4 docker image needs to be created first.

Comment (angerson) [reply tf_sig_build_dockerfiles/devel.usertools/gpu.bazelrc:47]: Right. We need those for remote-build-execution testing, but AFAIK the container should work as-is for building locally. The toolchain in remote_config, if I understand correctly, unnecessarily couples the RBE env variables with the c++ toolchain, which itself respects local env variables. So we could do some non-rbe testing with this as-is, but we'll need to have the crosstool updated eventually.

**Comment by angerson** [comment]
Thanks. Have you done any local testing with this container yet?

**Comment by nluehr** [comment]
I have run the pip and nonpip tests locally as described in [tf_sig_build_dockerfiles/README.md](https://github.com/tensorflow/build/blob/master/tf_sig_build_dockerfiles/README.md) on V100 and A100.

The only test failures were addressed in TF PRs [56821](https://github.com/tensorflow/tensorflow/pull/56821) and [56823](https://github.com/tensorflow/tensorflow/pull/56823). All should be passing now.

**Comment by angerson** [comment]
These dockerfiles have just been integrated directly into the main TF repository: https://github.com/tensorflow/tensorflow/tree/master/tensorflow/tools/tf_sig_build_dockerfiles

Can you recreate this PR over in the main TF repo? You can still use this PR if you want to, but we won't be able to merge it.

**Pull request closed without merging**